pkg/dirpublisher/dirpublisher.go

```diff
@@ -219,3 +221,128 @@ L1:
 	_ = os.RemoveAll(tmpDir)
 	return nil
 }
+
+func (dp *DirPublisher) CidToWeb3Storage(ctx context.Context, c cid.Cid) (cid.Cid, error) {
+	carReader, carWriter := io.Pipe()
+
+	jobs := []wpool.Job{
+		{
+			ID:   1,
+			Desc: "write car",
+			ExecFn: func(ctx context.Context) (interface{}, error) {
+				if err := car.WriteCar(ctx, dp.ipfsClient.Dag(), []cid.Cid{c}, carWriter); err != nil {
+					return nil, fmt.Errorf("writing car: %v", err)
+				}
+				_ = carWriter.Close()
+				return nil, nil
+			},
+		},
+		{
+			ID:   2,
+			Desc: "upload",
+			ExecFn: func(ctx context.Context) (interface{}, error) {
+				c, err := dp.web3Storage.PutCar(ctx, carReader)
+				if err != nil {
+					return nil, fmt.Errorf("uploading car: %v", err)
+				}
+				return c, nil
+			},
+		},
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	pool := wpool.New(2, rate.Inf)
+	go pool.GenerateFrom(jobs)
+	go pool.Run(ctx)
+	var res cid.Cid
+L:
+	for {
+		select {
+		case r, ok := <-pool.Results():
+			if !ok {
+				continue
+			}
+			if r.Err != nil {
+				return cid.Cid{}, fmt.Errorf("executing job %d, %s: %v", r.ID, r.Desc, r.Err)
+			}
+			if r.ID == 2 {
+				res = r.Value.(cid.Cid)
+			}
+		case <-pool.Done:
+			break L
+		}
+	}
+	log.Default().Printf("%s uploaded to web3.storage", res.String())
+	return res, nil
+}
+
+// RigsIndexToWeb3Storage creates a dagcbor index from the provided Rigs and adds it to web3.storage.
+func (dp *DirPublisher) RigsIndexToWeb3Storage(ctx context.Context, rigs []local.Rig) (cid.Cid, error) {
```

Comment:
@brunocalza and @jsign, I'm having some trouble in this code that writes some dagcbor and uploads it to web3.storage. Wondering if you can see anything wrong. You can follow my comments in the below function, it's pretty simple. Thanks.